### PR TITLE
feat(2.0 nodejs): Add `setStrongholdPassword` for `SecretManager`

### PR DESCRIPTION
**Pull request by marc2332**
Depends on https://github.com/iotaledger/iota-sdk/pull/1797 (sorry)

# Description of change

Enables support for `setStrongholdPassword` in Nodejs's `SecretManager`

## Type of change

- Enhancement

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Thoralf-M** [comment]
Why does it depend on https://github.com/iotaledger/iota-sdk/pull/1797? https://github.com/iotaledger/iota-sdk/pull/1807/commits/b67c8126899a72560d8d2dc0e42966da8f58db4a works alone?

**Comment by marc2332** [comment]
> Why does it depend on #1797? [b67c812](https://github.com/iotaledger/iota-sdk/commit/b67c8126899a72560d8d2dc0e42966da8f58db4a) works alone?

It's true that they don't depend on each other but it was easier for me to test this in Firefly, sorry, we are kind of blocked by the two other PRs, this way was easier :pray: 
I would review/merge the PRs in the order they were opened, sound good?

File: bindings/core/src/method_handler/secret_manager.rs [hunk 1]
```diff
@@ -135,6 +135,18 @@ where
...
+        SecretManagerMethod::SetStrongholdPassword { password } => {
+            if let Some(secret_manager) = secret_manager.downcast::<StrongholdSecretManager>() {
+                secret_manager.set_password(password).await?;
+                Response::Ok
+            } else if let Some(SecretManager::Stronghold(secret_manager)) = secret_manager.downcast::<SecretManager>() {
+                secret_manager.set_password(password).await?;
+                Response::Ok
+            } else {
+                return Err(iota_sdk::client::Error::SecretManagerMismatch.into());
+            }
```
Comment (Thoralf-M) [inline bindings/core/src/method_handler/secret_manager.rs]: I think this also works
```suggestion
            let stronghold = if let Some(secret_manager) = secret_manager.downcast::<StrongholdSecretManager>() {
                secret_manager
            } else if let Some(SecretManager::Stronghold(secret_manager)) = secret_manager.downcast::<SecretManager>() {
                secret_manager
            } else {
                return Err(iota_sdk::client::Error::SecretManagerMismatch.into());
            };
            stronghold.set_password(password).await?;
            Response::Ok
```

Comment (marc2332) [reply bindings/core/src/method_handler/secret_manager.rs]: Yeah, I can do it if you want. I just followed the same design as the method above ([store_mnemonic](https://github.com/iotaledger/iota-sdk/blob/d57dad9a35a7a169b119c65d17a2d0bee7712760/bindings/core/src/method_handler/secret_manager.rs#L126))

Comment (Thoralf-M) [reply bindings/core/src/method_handler/secret_manager.rs]: Could also be changed, but it's also not that important

**Review by Thoralf-M — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

File: bindings/nodejs/lib/types/secret_manager/bridge/secret-manager.ts [hunk 2]
```diff
@@ -70,3 +70,8 @@ export interface __SignSecp256k1EcdsaMethod__ {
 export interface __GetLedgerNanoStatusMethod__ {
     name: 'getLedgerNanoStatus';
 }
+
+export type __SetStrongholdPasswordMethod__ = {
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/secret_manager/bridge/secret-manager.ts]: I don't know much TS but all the others are interfaces, can it be as well for consistency or does it really need to be a type?

Comment (marc2332) [reply bindings/nodejs/lib/types/secret_manager/bridge/secret-manager.ts]: wew, fair point, I have no idea why I made it a type, changing this right away

File: bindings/nodejs/lib/types/secret_manager/bridge/secret-manager.ts [hunk 3]
```diff
@@ -70,3 +70,8 @@ export interface __SignSecp256k1EcdsaMethod__ {
 export interface __GetLedgerNanoStatusMethod__ {
     name: 'getLedgerNanoStatus';
 }
+
+export interface __SetStrongholdPasswordMethod__ = {
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/secret_manager/bridge/secret-manager.ts]:
```suggestion
export interface __SetStrongholdPasswordMethod__ {
```

Comment (marc2332) [reply bindings/nodejs/lib/types/secret_manager/bridge/secret-manager.ts]: Sorry again

Comment (thibault-martinez) [reply bindings/nodejs/lib/types/secret_manager/bridge/secret-manager.ts]: No worries at all :)

Comment (thibault-martinez) [reply bindings/nodejs/lib/types/secret_manager/bridge/secret-manager.ts]: But format is failing now :trollface:

**Review by thibault-martinez — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**